internal/receiver/smartagentreceiver/receiver_linux_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+
+package smartagentreceiver
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/signalfx/signalfx-agent/pkg/core/common/constants"
+	"github.com/signalfx/signalfx-agent/pkg/core/config"
+	"github.com/signalfx/signalfx-agent/pkg/monitors/collectd/uptime"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/extension/healthcheckextension"
+	"go.uber.org/zap"
+
+	"github.com/signalfx/splunk-otel-collector/internal/extension/smartagentextension"
+)
+
+var cleanUp = func() {
+	configuredCollectd = false
+}
+
+func TestSmartAgentReceiverDefaultCollectdConfig(t *testing.T) {
+	t.Cleanup(cleanUp)
+	cfg := Config{
+		ReceiverSettings: configmodels.ReceiverSettings{
+			TypeVal: typeStr,
+			NameVal: fmt.Sprintf("%s/%s", typeStr, "valid"),
+		},
+		monitorConfig: &uptime.Config{
+			MonitorConfig: config.MonitorConfig{
+				Type: "collectd/uptime",
+			},
+		},
+	}
+	r := NewReceiver(zap.NewNop(), cfg, consumertest.NewMetricsNop())
+
+	// Test whether default config is correctly loaded from smartagent extension factory.
+	// Note, that the below invocation of Start method is bound to fail since collectd
```

Comment:
Can this succeed if the collectd happens to be installed on the machine? Would that be a problem or we don't care if it succeeds?

Reply:
If collectd is already running on the machine when the receiver starts up, it will just create a new instance of collectd.